src/utils/readers/scientific.py

```
@@ -28,182 +46,269 @@
-
-        if len(var_names) > 30:
-            lines.append(f"  ... and {len(var_names) - 30} more variables")
-
-        text = "\n".join(lines)
-        logger.debug(f"Extracted metadata from MAT file {file_path.name}")
-        return text
-
+        return _parse_mat(path, path.name)
     except Exception as e:  # Intentional catch-all: scipy.io raises library-specific errors
-        raise FileReadError(f"Failed to read MAT file {file_path}: {e}") from e
+        raise FileReadError(f"Failed to read MAT file {path}: {e}") from e
```

Ruff (0.15.13)

[warning] Avoid specifying long messages outside the exception class

(TRY003)